docs/how-to/rccl-usage-tips.rst

```diff
@@ -82,22 +82,198 @@ set the HSA environment variable as follows:
+
+  - In the SPX mode, workgroups launched to the device are distributed
+    round-robin to the XCDs in the device. Meaning that the programmer cannot
+    have explicit control over which XCD a workgroup is assigned to.
+
+  - In the CPX mode, workgroups are launched to a single XCD, meaning the
+    programmer has explicit control over work placement onto the XCDs.
+  
+- Memory partition modes 
+
+  - In NPS1 mode (compatible with CPX and SPX), the entire memory is accessible
+    to all XCDs.
+
+  - In NPS4 mode (compatible with CPX) Each memory quadrant of the memory is
+    directly visible to the logical devices in it's quadrant. An XCD can still
+    access all portions of memory through multi-GPU programming techniques.
+
+For more information, see :doc:`Dynamically change GPU partition modes <rocm:how-to/dynamic_partition_modes>`
+page. The MI300 CPX mode can be accessed using the following :doc:`rocm-smi:index`
+commands.
+
+.. code-block:: shell
+
+   sudo rmmod amdgpu
+   sudo rmmod ast
+   sudo modprobe amdgpu
+   rocm-smi --setcomputepartition CPX
+ 
+   ## rocm-smi should list 64 GPUs (devices 0 to 63)
+   rocm-smi
+
+This setting can be also configured with sysfs, as shown in the following example.
+
+.. code-block:: shell
+
+   $ cat /sys/bus/pci/devices/../current_memory_partition
+   NPS1
+   $ cat /sys/bus/pci/devices/../current_compute_partition
+   SPX
+   $ echo NPS4 > /sys/bus/pci/devices/../current_memory_partition
+   $ rmmod amdgpu
+   $ modprobe amdgpu
+   $ cat /sys/bus/pci/devices/../current_memory_partition
+   NPS4
+   $ cat /sys/bus/pci/devices/../current_compute_partition
+   CPX
+
+RCCL performance with CPX and NPS4
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To run RCCL allreduce on 64 GPUs with CPX+NPS4 mode on the MI300X, use this
+example:
+
+.. code-block:: shell
+
+   mpirun -np 64 --bind-to numa rccl-tests/build/all_reduce_perf -b 8 -e 1G -f 2 -g 1
+
+To run RCCL allreduce on 8 GPUs in the same OAM with CPX+NPS4 mode on the
+MI300X, use this example:
+
+.. code-block:: shell
+
+   export ROCR_VISIBLE_DEVICES=0,1,2,3,4,5,6,7
+
+   mpirun -np 8 --bind-to numa rccl-tests/build/all_reduce_perf -b 8 -e 1G -f 2 -g 1
+
+RCCL delivers improved allreduce performance in CPX mode for TP=8 (8 GPUs in
+the same OAM) on the MI300X.
+
+.. code-block:: shell
+
+   export HIP_FORCE_DEV_KERNARG=1
+   export RCCL_MSCCLPP_THRESHOLD=1073741824
+
+   export MSCCLPP_READ_ALLRED=1 
+   export ROCR_VISIBLE_DEVICES=0,1,2,3,4,5,6,7
+
+   mpirun -np 8 --bind-to numa rccl-tests/build/all_reduce_perf -b 32 -e 1G -f 2 -g 1 -G 2 -w 20 -n 50
+
+Here are the benchmark results for in-place (where the output buffer is used as
+the input buffer) and out-of-place allreduce bus bandwidth.
+
+.. figure:: ../data/how-to/rccl-usage-tips/in-place_allreduce.png
+    :alt: In-place allreduce benchmark results
+    :align: center
+
+.. figure:: ../data/how-to/rccl-usage-tips/out-of-place_allreduce.png
+    :alt: Out-of-place allreduce benchmark results
```

Comment:
The graphs still say reduce in the legends. This should be corrected.

Reply:
Oh, sorry for that.

Reply:
Fixed.